src/Router/Route.php

```diff
@@ -133,6 +151,14 @@ public function setOptions(array $options)
     }
 
     /**
+     * @param string $name
+     */
+    public function setName($name)
```

Comment:
I'm going to remove the setter on merge. Essentially, if we set it after-the-fact, the name is irrelevant for the primary use case of consuming it with a router for URI assembly. As such, making it mutable actually is counter-productive.